google/resource_cloudiot_registry.go

```diff
@@ -0,0 +1,411 @@
+	mqttDisabled          = "MQTT_DISABLED"
+	httpStateUnspecified  = "HTTP_STATE_UNSPECIFIED"
+	httpEnabled           = "HTTP_ENABLED"
+	httpDisabled          = "HTTP_DISABLED"
+	unspecifiedCertFormat = "UNSPECIFIED_PUBLIC_KEY_CERTIFICATE_FORMAT"
+	x509CertificatePEM    = "X509_CERTIFICATE_PEM"
+)
+
+func resourceCloudiotRegistry() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceCloudiotRegistryCreate,
+		Update: resourceCloudiotRegistryUpdate,
+		Read:   resourceCloudiotRegistryRead,
+		Delete: resourceCloudiotRegistryDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceCloudiotRegistryStateImporter,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: linkDiffSuppress,
+			},
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"region": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"event_notification_configs": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"pubsub_topic_name": &schema.Schema{
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: compareSelfLinkOrResourceName,
+						},
+					},
+				},
+			},
+			"state_notification_config": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: false,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"pubsub_topic_name": &schema.Schema{
+							Type:             schema.TypeString,
+							Required:         true,
+							DiffSuppressFunc: compareSelfLinkOrResourceName,
+						},
+					},
+				},
+			},
+			"mqtt_config": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: false,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"mqtt_enabled_state": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice(
+								[]string{mqttStateUnspecified, mqttEnabled, mqttDisabled}, false),
+						},
+					},
+				},
+			},
+			"http_config": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: false,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"http_enabled_state": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice(
+								[]string{httpStateUnspecified, httpEnabled, httpDisabled}, false),
+						},
+					},
+				},
+			},
+			"credentials": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: false,
+				MaxItems: 10,
```

Comment:
Where does the max of 10 come from?

Reply:
I found the max 10 in the documentation of [DeviceRegistry](https://cloud.google.com/iot/docs/reference/rest/v1/projects.locations.registries#DeviceRegistry) for the `credentials` field: "No more than 10 credentials can be bound to a single registry at a time.". Hope I got it right.